src/nvim/menu.c

```diff
@@ -679,7 +677,9 @@ static dict_T *menu_get_recursive(const vimmenu_T *menu, int modes)
   tv_dict_add_nr(dict, S_LEN("hidden"), menu_is_hidden(menu->dname));
 
   if (menu->mnemonic) {
-    mnemonic_len = utf_char2bytes(menu->mnemonic, (u_char *)buf);
+    int mnemonic_len;
+    char_u buf[8];  // > max value of char2bytes
```

Comment:
`utf_char2bytes` says 6 is the maximum bytes it would need.

```
/// @return Number of bytes (1-6). Does not include composing characters.
```

Why not default initialize to 0 and then let `utf_char2bytes` overwrite what it needs?

```c
char buf[7] = {0};  // Max return from utf_char2bytes + 1 for NUL
utf_char2bytes(menu->mnemonic, (char_u *)buf);
tv_dict_add_str(dict, S_LEN("shortcut"), buf);
```